packages/jbrowse-plugin-apollo/src/LinearApolloDisplay/components/ApolloDetails.tsx

```diff
@@ -148,20 +185,78 @@ export const ApolloDetails = observer(
       end,
       assemblyId: assembly,
     } = selectedFeature
-    // TODO: make this more generic
+    let refName = refSeq
+    const { assemblyManager } = session
+
+    if (assemblyManager.get(assembly)?.refNameAliases) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const refNames = assemblyManager.get(assembly)!.refNameAliases!
+      Object.keys(refNames).forEach((key) => {
+        if (key === refSeq) {
+          refName = refNames[key]
+        }
+      })
+    }
```

Comment:
There's a method on the assembly manager called `getCanonicalRefName` that we can use here.

```suggestion
    const { assemblyManager } = session

    const refName =
      assemblyManager.get(assembly)?.getCanonicalRefName(assembly) || refSeq
```

Reply:
small correction  `assemblyManager.get(assembly)?.getCanonicalRefName(refSeq) || refSeq`